shell/src/main/java/org/jtaccuino/jshell/ReactiveJShell.java

```diff
@@ -103,26 +103,32 @@ public EvaluationResult eval(String string) {
         }
 
         if (snippetEventsCurrentSnippets.stream().allMatch(event -> Snippet.Kind.ERRONEOUS != event.snippet().kind() && Snippet.Status.VALID == event.status())) {
-            var lastEvent = snippetEventsCurrentSnippets.getLast();
-            if (lastEvent.snippet() instanceof VarSnippet v) {
-                var varName = v.name();
-                var varType = sourceCodeAnalysis().analyzeType(varName,
-                        varName.length());
-                var varValue = lastEvent.value();
-                if (varName.startsWith("$")) {
-                    return new EvaluationResult(snippetEventsCurrentSnippets, snippetEventsInfluencedSnippets, ResultStatus.SUCCESS, Optional.of(varValue), Optional.of(varType));
+            var mayBeLastEvent = Optional.ofNullable(snippetEventsCurrentSnippets.isEmpty() ? null : snippetEventsCurrentSnippets.getLast());
+            return mayBeLastEvent.map(lastEvent
+                    -> switch (lastEvent.snippet()) {
+                case VarSnippet v when v.name().startsWith("$") -> {
+                    var varName = v.name();
+                    var varType = sourceCodeAnalysis().analyzeType(varName,
+                            varName.length());
+                    var varValue = lastEvent.value();
+                    yield new EvaluationResult(snippetEventsCurrentSnippets, snippetEventsInfluencedSnippets,
+                    ResultStatus.SUCCESS, Optional.of(varValue), Optional.of(varType));
                 }
-            }
-            if (lastEvent.snippet() instanceof ExpressionSnippet e) {
-                var varName = e.name();
-                var varType = sourceCodeAnalysis().analyzeType(varName,
-                        varName.length());
-                var varValue = lastEvent.value();
-                return new EvaluationResult(snippetEventsCurrentSnippets, snippetEventsInfluencedSnippets, ResultStatus.SUCCESS, Optional.of(varValue), Optional.of(varType));
-            }
-            return new EvaluationResult(snippetEventsCurrentSnippets, snippetEventsInfluencedSnippets, ResultStatus.SUCCESS, Optional.empty(), Optional.empty());
+                case ExpressionSnippet e -> {
+                    var varName = e.name();
+                    var varType = sourceCodeAnalysis().analyzeType(varName,
+                            varName.length());
+                    var varValue = lastEvent.value();
+                    yield new EvaluationResult(snippetEventsCurrentSnippets, snippetEventsInfluencedSnippets,
+                    ResultStatus.SUCCESS, Optional.of(varValue), Optional.of(varType));
+                }
+                default ->
+                    new EvaluationResult(snippetEventsCurrentSnippets, snippetEventsInfluencedSnippets,
+                    ResultStatus.SUCCESS, Optional.empty(), Optional.empty());
+            }).orElse(new EvaluationResult(snippetEventsCurrentSnippets, snippetEventsInfluencedSnippets,
```

Comment:
using orElseGet prevent unnecessary creation of the nondescript SUCCESS EvaluationResult
```suggestion
            }).orElseGet(() -> new EvaluationResult(snippetEventsCurrentSnippets, snippetEventsInfluencedSnippets,
```